system/Entity/EntityInterface.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Entity;
+
+/**
+ * Entity encapsulation, for use with CodeIgniter\Model
+ *
+ * @see \CodeIgniter\Entity\EntityTest
+ */
+interface EntityInterface
+{
+    /**
+     * Allows filling in Entity parameters during construction.
+     */
+    public function __construct(?array $data = null);
```

Comment:
Unless there is a very strong reason to do so, constructors should not be included in the interface.